mutation/index.ts

```diff
@@ -72,11 +76,14 @@ const mutation = (<Data, Error>() =>
           }
           return data
         } catch (error) {
-          // If it's reset after the mutation, we don't broadcast any state change.
+          // If it's reset after the mutation, we don't broadcast any state change
+          // or throw because it's discarded.
           if (ditchMutationsUntilRef.current <= mutationStartedAt) {
             setState({ error: error as Error, isMutating: false })
             options.onError?.(error as Error, serializedKey, options)
```

Comment:
Is `onError` always called regardless of `throwOnError`?

> should call `onError` event and skip throwing the error when `throwOnError` is disabled

https://github.com/vercel/swr/pull/2182/files#diff-665896fca56ef021ec0b6b13ffb986551f53f66c78ba58ac25ab8744084d3daeR168

I guess `onError` is not called with `throwOnError: true` from the test description.